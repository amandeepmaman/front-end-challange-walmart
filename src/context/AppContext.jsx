import React, { createContext, useState } from 'react';

export const AppContext = createContext();

export const AppProvider = ({ children }) => {
  const [activePage, setActivePage] = useState('apps');
  const [isChatOpen, setIsChatOpen] = useState(false);
  const [messages, setMessages] = useState([]);
  const [searchQuery, setSearchQuery] = useState('');

  const resetChat = () => {
    setMessages([]);
  };

  const sendMessage = (message) => {
    const newMessage = {
      id: Date.now(),
      text: message,
      sender: 'user',
    };
    
    setMessages(prev => [...prev, newMessage]);
    
    // Simulate bot response
    setTimeout(() => {
      const botResponses = [
        "I'm a chatbot. How can I help you?",
        "That's interesting. Tell me more.",
        "I'm still learning. Can you rephrase that?",
        "Thanks for your message!",
        "I'll need to check on that for you."
      ];
      const botMessage = {
        id: Date.now() + 1,
        text: botResponses[Math.floor(Math.random() * botResponses.length)],
        sender: 'bot',
      };
      setMessages(prev => [...prev, botMessage]);
    }, 1000);
  };

  return (
    <AppContext.Provider
      value={{
        activePage,
        setActivePage,
        isChatOpen,
        messages,
        resetChat,
        setIsChatOpen,
        sendMessage,
        searchQuery,
        setSearchQuery
      }}
    >
      {children}
    </AppContext.Provider>
  );
};