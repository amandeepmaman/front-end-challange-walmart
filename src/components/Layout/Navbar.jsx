import React, { useContext } from 'react';
import { AppContext } from '../../context/AppContext.jsx';

const Navbar = () => {
  const { searchQuery, setSearchQuery } = useContext(AppContext);

  return (
    <nav className="navbar" data-testid="navbar">
      <div className="logo">Logo</div>
      <input
        type="text"
        placeholder="Search..."
        value={searchQuery}
        onChange={(e) => setSearchQuery(e.target.value)}
        className="search-bar"
      />
    </nav>
  );
};

export default Navbar;