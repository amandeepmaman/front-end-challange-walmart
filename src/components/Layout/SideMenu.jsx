import React, { useContext } from 'react';
import { AppContext } from '../../context/AppContext.jsx';

const SideMenu = () => {
  const { activePage, setActivePage } = useContext(AppContext);

  return (
    <div className="side-menu" data-testid="side-menu">
      <button 
        className={`menu-item ${activePage === 'apps' ? 'active' : ''}`}
        onClick={() => setActivePage('apps')}
      >
        Apps
      </button>
      <button 
        className={`menu-item ${activePage === 'documents' ? 'active' : ''}`}
        onClick={() => setActivePage('documents')}
      >
        Documents
      </button>
    </div>
  );
};

export default SideMenu;