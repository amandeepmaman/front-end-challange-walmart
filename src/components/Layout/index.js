import React from "react";
import Navbar from "./Navbar";
import SideMenu from "./SideMenu";

const Layout = ({ children }) => {
  return (
    <div className="app">
      <Navbar />
      <div className="main-content">
        <SideMenu />
        <div className="content-area">{children}</div>
      </div>
    </div>
  );
};

export default Layout;
