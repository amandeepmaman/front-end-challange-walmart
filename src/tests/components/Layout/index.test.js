// __tests__/components/Layout/index.test.js
import React from "react";
import { render, screen } from "@testing-library/react";
import Layout from "../../../components/Layout";
import { AppProvider } from "../../../context/AppContext";

describe("Layout Component", () => {
  it("renders Navbar and SideMenu with children", () => {
    render(
      <AppProvider>
        <Layout>
          <div data-testid="test-child">Test Content</div>
        </Layout>
      </AppProvider>
    );

    // Verify Navbar is rendered
    expect(screen.getByText("Logo")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Search...")).toBeInTheDocument();

    // Verify SideMenu is rendered
    expect(screen.getByText("Apps")).toBeInTheDocument();
    expect(screen.getByText("Documents")).toBeInTheDocument();

    // Verify children are rendered
    expect(screen.getByTestId("test-child")).toBeInTheDocument();
    expect(screen.getByText("Test Content")).toBeInTheDocument();
  });

  it.skip("matches snapshot", () => {
    const { asFragment } = render(
      <AppProvider>
        <Layout>
          <div>Test Content</div>
        </Layout>
      </AppProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
