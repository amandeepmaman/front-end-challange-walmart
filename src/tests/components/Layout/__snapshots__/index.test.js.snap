// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <nav
      class="navbar"
    >
      <div
        class="logo"
      >
        Logo
      </div>
      <input
        class="search-bar"
        placeholder="Search..."
        type="text"
        value=""
      />
    </nav>
    <div
      class="main-content"
    >
      <div
        class="side-menu"
      >
        <button
          class="menu-item active"
        >
          Apps
        </button>
        <button
          class="menu-item "
        >
          Documents
        </button>
      </div>
      <div
        class="content-area"
      >
        <div>
          Test Content
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
